nexus_constructor/model/component.py

```diff
@@ -243,6 +243,7 @@ def add_rotation(
             type=ValueTypes.DOUBLE,
         ),
         target_pos: int = -1,
+        offset: QVector3D = QVector3D(),
```

Comment:
Ahh!  So here it's not an issue for now only because we aren't triggering it, which we may down the line*.  Here's a simple case where it shows the behaviour of a mutable object.  Like so:

>>> def somemethod(test=QVector3D()):
...     print(test)
...     test.setX(1)
... 
>>> somemethod()
PySide6.QtGui.QVector3D(0.000000, 0.000000, 0.000000)
>>> somemethod()
PySide6.QtGui.QVector3D(1.000000, 0.000000, 0.000000)

*Actually, the way the vector is used we never will, but I get the point that it's a bad idea!